src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -2,18 +2,35 @@
 
 using System;
 using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Build;
 using Microsoft.VisualStudio.ProjectSystem.Input;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
 {
-    internal abstract class AbstractGenerateNuGetPackageCommand : AbstractSingleNodeProjectCommand
+    internal abstract class AbstractGenerateNuGetPackageCommand : AbstractSingleNodeProjectCommand, IVsUpdateSolutionEvents
     {
         private readonly IProjectThreadingService _threadingService;
+        private readonly IVsSolutionBuildManager2 _buildManager;
+        private readonly GeneratePackageOnBuildPropertyProvider _generatePackageOnBuildPropertyProvider;
 
-        protected AbstractGenerateNuGetPackageCommand(UnconfiguredProject unconfiguredProject, IProjectThreadingService threadingService)
+        protected AbstractGenerateNuGetPackageCommand(
+            UnconfiguredProject unconfiguredProject,
+            IProjectThreadingService threadingService,
+            SVsServiceProvider serviceProvider,
+            GeneratePackageOnBuildPropertyProvider generatePackageOnBuildPropertyProvider)
         {
+            Requires.NotNull(serviceProvider, nameof(serviceProvider));
+            Requires.NotNull(generatePackageOnBuildPropertyProvider, nameof(generatePackageOnBuildPropertyProvider));
+
             UnconfiguredProject = unconfiguredProject;
             _threadingService = threadingService;
+            _generatePackageOnBuildPropertyProvider = generatePackageOnBuildPropertyProvider;
+
+            uint cookie;
+            _buildManager = serviceProvider.GetService<IVsSolutionBuildManager2, SVsSolutionBuildManager>();
```

Comment:
We should not do this on the constructor, it's a great way to cause deadlocks. Please implement `OnceInitializedOnceDestroyed` or `OnceInitializedOnceDestroyedAsync` and do this and other initialization there.